Android uses gcc -shared, etc as usual - but it is alarming how many
places in autoconf/libtool work on matching specific arch triples,
rather than actually probing for functionality...

--- libtool-2.2.6b/libltdl/m4/libtool.m4~	Thu Jun 24 08:24:09 2010
+++ libtool-2.2.6b/libltdl/m4/libtool.m4	Sat Jun 26 22:32:11 2010
@@ -2452,7 +2452,10 @@
   need_version=no
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
   soname_spec='${libname}${release}${shared_ext}$major'
-  finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
+  case $host_os in
+  *android*) ;;
+  *) finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir' ;;
+  esac
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
   # Some binutils ld are patched to set DT_RUNPATH
@@ -2471,19 +2474,27 @@
   # before this can be enabled.
   hardcode_into_libs=yes
 
-  # Append ld.so.conf contents to the search path
-  if test -f /etc/ld.so.conf; then
-    lt_ld_extra=`awk '/^include / { system(sprintf("cd /etc; cat %s 2>/dev/null", \[$]2)); skip = 1; } { if (!skip) print \[$]0; skip = 0; }' < /etc/ld.so.conf | $SED -e 's/#.*//;/^[	 ]*hwcap[	 ]/d;s/[:,	]/ /g;s/=[^=]*$//;s/=[^= ]* / /g;/^$/d' | tr '\n' ' '`
-    sys_lib_dlsearch_path_spec="/lib /usr/lib $lt_ld_extra"
-  fi
+  case $host_os in
+  *android*)
+    sys_lib_dlsearch_path_spec="/system/lib /lib $lt_ld_extra"
+    dynamic_linker='Android linker'
+    ;;
+  *)
+    # Append ld.so.conf contents to the search path
+    if test -f /etc/ld.so.conf; then
+      lt_ld_extra=`awk '/^include / { system(sprintf("cd /etc; cat %s 2>/dev/null", \[$]2)); skip = 1; } { if (!skip) print \[$]0; skip = 0; }' < /etc/ld.so.conf | $SED -e 's/#.*//;/^[	 ]*hwcap[	 ]/d;s/[:,	]/ /g;s/=[^=]*$//;s/=[^= ]* / /g;/^$/d' | tr '\n' ' '`
+      sys_lib_dlsearch_path_spec="/lib /usr/lib $lt_ld_extra"
+    fi
 
-  # We used to test for /lib/ld.so.1 and disable shared libraries on
-  # powerpc, because MkLinux only supported shared libraries with the
-  # GNU dynamic linker.  Since this was broken with cross compilers,
-  # most powerpc-linux boxes support dynamic linking these days and
-  # people can always --disable-shared, the test was removed, and we
-  # assume the GNU/Linux dynamic linker is in use.
-  dynamic_linker='GNU/Linux ld.so'
+    # We used to test for /lib/ld.so.1 and disable shared libraries on
+    # powerpc, because MkLinux only supported shared libraries with the
+    # GNU dynamic linker.  Since this was broken with cross compilers,
+    # most powerpc-linux boxes support dynamic linking these days and
+    # people can always --disable-shared, the test was removed, and we
+    # assume the GNU/Linux dynamic linker is in use.
+    dynamic_linker='GNU/Linux ld.so'
+    ;;
+  esac
   ;;
 
 netbsd*)

